projects/solar/test/test_sense.c

```diff
@@ -0,0 +1,177 @@
+#include "delay.h"
+#include "interrupt.h"
+#include "log.h"
+#include "ms_test_helpers.h"
+#include "sense.h"
+#include "soft_timer.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+#define TEST_SENSE_PERIOD_US 50000  // 50ms, should be okay
+
+static SenseSettings test_settings = {
+  .sense_period_us = TEST_SENSE_PERIOD_US,
+};
+
+static uint8_t s_times_data_store_done_called = 0;
+
+static uint8_t s_times_callback_called = 0;
+static void *s_callback_context = NULL;
+
+static uint8_t s_times_callback_2_called = 0;
+static void *s_callback_2_context = NULL;
+
+// We mock |data_store_done| to make sure it's being called without relying on data_store behaviour.
+StatusCode TEST_MOCK(data_store_done)(void) {
+  s_times_data_store_done_called++;
+  return STATUS_CODE_OK;
+}
+
+static void prv_callback(void *context) {
+  s_times_callback_called++;
+  s_callback_context = context;
+}
+
+static void prv_callback_2(void *context) {
+  s_times_callback_2_called++;
+  s_callback_2_context = context;
+}
+
+void setup_test(void) {
+  event_queue_init();
```

Comment:
My rationale for leaving it in is that technically, whether it internally uses the event queue or not is an implementation detail of the module (especially if we're just saying "requires the event queue to be initialized"), and so technically the test shouldn't know that implementation detail. (It's not a perfect argument since the tests are somewhat privy to the implementation details of the module anyways, but I think it makes sense to leave it in.)